src/components/tree-view/TreeViewItem.tsx

```diff
@@ -131,7 +151,7 @@ export const TreeViewItem = <T,>({
               <span
                 onClick={(e) => {
                   e.stopPropagation();
-                  void handleClick();
```

Comment:
We can remove totally the function `handleClick`, inside `handleOver` you can replace `void handleClick();` by `node.open();`. We don't need to memoized `handleOver` neither, it can directly go inside the `useEffect` under I think.